client/client.cpp

```diff
@@ -1,23 +1,78 @@
+#include <iostream>
+#include <sodium.h>
+#include <thrift/protocol/TBinaryProtocol.h>
+#include <thrift/transport/TSocket.h>
+#include <thrift/transport/TTransportUtils.h>
+
+#include "../constants/network.h"
 #include "../crypto/encryption_engine.h"
+#include "../gen-cpp/Operation_types.h"
+#include "../gen-cpp/RPC.h"
 #include "../host/redis.h"
-#include <iostream>
-using namespace std;
+
+using namespace apache::thrift;
+using namespace apache::thrift::protocol;
+using namespace apache::thrift::transport;
+
+Operation genRandOperation() {
+	float r = (float) rand() / RAND_MAX;
+	int key = rand() % KEY_MAX;
+  
+	Operation op;
+    
+	op.__set_op(r < 0.5 ? "PUT" : "GET");
+  op.__set_key(std::string(std::to_string(key)));
+	if(op.op == "GET") {
+		char value[VALUE_SIZE];
+		randombytes_buf(value, VALUE_SIZE);
+		op.__set_value(std::string(value));
+	}
+	return op;
+}
+
+void client(std::vector <float>* latencies) {
+	std::shared_ptr<TTransport> socket(new TSocket(HOST_IP, HOST_PORT));
+	std::shared_ptr<TTransport> transport(new TBufferedTransport(socket));
+	std::shared_ptr<TProtocol> protocol(new TBinaryProtocol(transport));
+	RPCClient client(protocol);
+	Operation op;
+
+	try {
+		transport->open();
+
+		for (int i = 0; i < 10; ++i) {
+			op = genRandOperation();
+			std::string val;
+			client.access(val, op);
+		}
+		
+		transport->close();
+	} catch (TException& tx) {
+    std::cout << "ERROR: " << tx.what() << std::endl;
+  }
+}
+
 
 int main(int argc, char *argv[]) {
-	encryption_engine encryption_engine_;
-	redisCli rd;
-	unsigned char cipher_text[4096];
-	auto encLen = encryption_engine_.encryptNonDeterministic("30", cipher_text);
-	string val((const char *)cipher_text, encLen);
-	cout << "Encrypted value is: " << val << endl;
-	rd.put("1", val);
-	cout << "Redis get encrypted for key 1: " << rd.get("1") << endl;
-    cout << "Decrypted value for key 1: " << encryption_engine_.decryptNonDeterministic(rd.get("1")) << endl;
+	std::unique_ptr<std::vector<float>> latencies;
+	client(latencies.get());
```

Comment:
Is there a reason you're using a pointer here? Just create the vector and change the signature of client to `std::vector<float> &latencies` (reference)

Reply:
Yep that makes sense. I have yet to work on the latency part yet, the pointer is a remanent of an older implementation of client for ORTOA.

Reply:
Latencies have now been addressed, closing this comment